src/deprecate.jl

```diff
@@ -0,0 +1,2 @@
+Base.@deprecate qr_compact_pullback!(args...) qr_pullback!(args...) false
+Base.@deprecate lq_compact_pullback!(args...) lq_pullback!(args...) false
```

Comment:
I assume deprecated methods are exported? Hence, the methods we deprecate are "more visible" than the ones that replace them. How should one deprecate non-exported public functions?

(I am fine with a breaking version after the current bunch of PRs, and not yet caring about deprecating at this stage)

Reply:
They are, but there is a final argument to the macro `false` to disable that behavior, so these methods aren't exported in this case.